frontend/src/pages/tokens/index.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+	<div>
+		<Title>CCDScan | Tokens</Title>
+
+		<Table>
+			<TableHead>
+				<TableRow>
+					<TableTh>Token Id</TableTh>
+					<TableTh v-if="breakpoint >= Breakpoint.LG">Contract</TableTh>
+					<TableTh v-if="breakpoint >= Breakpoint.SM">{{
+						breakpoint >= Breakpoint.LG ? 'Metadata Url' : 'Metadata'
+					}}</TableTh>
+					<TableTh align="right"> Supply </TableTh>
+				</TableRow>
+			</TableHead>
+			<TableBody>
+				<TableRow
+					v-for="token in tokens"
+					:key="token.contractIndex + token.tokenId"
+				>
+					<TableTd>
+						<TokenLink :data="token.tokenId" :url="token.metadataUrl" />
+					</TableTd>
+					<TableTd>
+						{{ token.contractIndex }} / {{ token.contractSubIndex }}
```

Comment:
We should display the "TokenAddress" somewhere https://github.com/Concordium/concordium-base/blob/70c1c88250dc65a56bca46965c2df9a278a5703e/rust-src/concordium_base/src/cis2_types.rs#L91

This contains index + subindex + tokenId in a single copyable string. See https://github.com/Concordium/concordium-base/blob/70c1c88250dc65a56bca46965c2df9a278a5703e/rust-src/concordium_base/src/cis2_types.rs#L1195 for examples.

Reply:
We attempted this. However there seems to be now way of knowing the type of token id. All the responses we get back from the node return token id as `string`. The actual type of token if `u128`, `u64` etc. Cannot be inferred from the token events. 

Can you help us in finding a way to decipher the type of Token Id?

Reply:
I don't understand. For the token address it does not matter whether token id is a u128 or u16 or () internally by the contract.

The token id format is specified by the standard and it serialization is fixed. What you get back is probably a hex string with the serialization, and that is all you need to construct the token address.